src/structures/PermissionOverwrites.js

```diff
@@ -64,6 +64,32 @@ class PermissionOverwrites {
   toJSON() {
     return Util.flatten(this);
   }
+
+  /**
+   * Resolves a PermissionOverwrites from options or from another overwrite.
+   * @param {OverwriteData|PermissionOverwrites} overwrite Overwrite to create from
+   * @param {Guild} guild Guild to base role data on
+   * @returns {PermissionOverwrites}
+   */
+  static resolve(overwrite, guild) {
+    const role = guild.roles.resolve(overwrite.id);
+    let id;
+    let type;
+    if (role) {
```

Comment:
if overwrite is a PermissionOverwrites, or if the user passes correct data for id/type (which `type` isn't included in the OverwriteData typedef, and it probably should be), then those shouldn't be re-resolved.

```js
if (typeof overwrite.id === 'string' && ['role', 'member'].includes(overwrite.type)) {
  id = overwrite.id;
  type = overwrite.type;
} else {
  const role = guild.roles.resolve(overwrite.id);
  if (role) {
    id = role.id;
    type = 'role';
  } else {
    id = guild.client.users.resolveID(overwrite.id);
    type = 'member';
  }
}
```